src/ast/ast_validations.c

```diff
@@ -1835,14 +1853,18 @@ AST_Validation AST_Validate_Query(const cypher_parse_result_t *result) {
 	// Verify that the query does not contain any expressions not in the
 	// RedisGraph support whitelist
 	if(CypherWhitelist_ValidateQuery(root) != AST_VALID) return AST_INVALID;
-
+	
+	AST_Validation res;
 	const cypher_astnode_t *body = cypher_ast_statement_get_body(root);
 	cypher_astnode_type_t body_type = cypher_astnode_type(body);
 	if(body_type == CYPHER_AST_CREATE_NODE_PROPS_INDEX    ||
 	   body_type == CYPHER_AST_CREATE_PATTERN_PROPS_INDEX ||
 	   body_type == CYPHER_AST_DROP_PROPS_INDEX) {
-		// Index operation; validations are handled elsewhere.
-		return AST_VALID;
+		// Index operation
+		rax *defined_aliases = raxNew();
+		res = _Validate_Aliases_DefinedInClause(body, defined_aliases);
```

Comment:
you can calll _AST_CreateIndex_GetDefinedIdentifiers directly

Reply:
Sorry but I did not understand your idea, probably I'm missing something.
I think that I need to call _Validate_Aliases_DefinedInClause() because it iterates the referred identifiers searching them in the defined identifiers that are obtained by _AST_CreateIndex_GetDefinedIdentifiers().
Or do you mean to rewrite the _AST_CreateIndex_GetDefinedIdentifiers() to make that iteration and validations? 
Thanks.